lohsuan/puppy_sponsor
# Refactor project code format and add CI configs

## Description
- The original code of the project has no fixed format and is messy, so I added `eslint` and `prettier` for unified formatting.
  So you can check & fix code format by using `pnpm lint`, `pnpm lint:fix` and `pnpm format` after this PR.

- In addition, I also added the `dockerfile` that will be used for deployment, as well as the CI configuration file.

- And move asset files into `src/assets` folder. (json, css, images)

Also, since the CI config file has not been merged into the main branch, the CI check results cannot be seen here.
If it runs failed later, I will open another PR to fix it.

## How to verify?
Since almost all the files are reformatted, it is recommended not to view them one by one, as you may not be able to read them all.

- run dev server to see if it looks OK (go to every pages, click every buttons, etc).
- run production build and `pnpm preview` to see if the production builds normally.
